### PR TITLE
CIF and XYZ files in prompts

### LINKED ISSUE #393: CIF and XYZ files in prompts
examples https://gist.github.com/kjappelbaum/a5f855945582c3f00c4bb245e5432bfe

### PR DESCRIPTION
**Pull request by kjappelbaum**
Fixes #393

i initially thought that incorporating it directly in the CSV will not work (because I was afraid that it would confuse the parser). However, it worked very well wherefore I did not touch the sampling script but only pre-process data. 

This includes also quite conservative filtering out of large structures

### CODE REVIEW

**Comment by kjappelbaum** [comment]
Screenshot 2023-10-20 at 07 07 22

File: data/tabular/smiles_to_3d/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      author={Ruddigkeit, Lars and Van Deursen, Ruud and Blum, Lorenz C and Reymond, Jean-Louis},
+      journal={Journal of chemical information and modeling},
+      volume={52},
+      number={11},
+      pages={2864--2875},
+      year={2012},
+      publisher={ACS Publications}}
+templates:
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {xyz__names__noun} of a {#molecule|compound|chemical!} with SMILES {SMILES#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:81]: Add a `the`?
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {xyz__names__noun} of the {#molecule|compound|chemical!} with the SMILES {SMILES#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {xyz#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol2000__names__noun} of a {#molecule|compound|chemical!} with SMILES {SMILES#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:84]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol2000__names__noun} of the {#molecule|compound|chemical!} with the SMILES {SMILES#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {mol2000#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol3000__names__noun} of a {#molecule|compound|chemical!} with SMILES {SMILES#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:87]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol3000__names__noun} of the {#molecule|compound|chemical!} with the SMILES {SMILES#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {mol3000#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of a {#molecule|compound|chemical!} with {xyz__names__noun} {xyz#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:90]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with the {xyz__names__noun} {xyz#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {SMILES#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of a {#molecule|compound|chemical!} with {mol2000__names__noun} {mol2000#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:93]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with the {mol2000__names__noun} {mol2000#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {SMILES#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of a {#molecule|compound|chemical!} with {mol3000__names__noun} {mol3000#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:96]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with the {mol3000__names__noun} {mol3000#}?
```

File: data/text_sampling/text_sampling.py [hunk 7]
```diff
@@ -969,8 +970,8 @@ def apply_sampling_and_export(
     # path_data_dir = path_data_dir[index:]
 
     for path in path_data_dir:
-        if path.find("data/tabular/bicerano_dataset") != -1:
-            continue  # needs the units fixed or templates copied over
+        # if "smiles_to_3d" not in path:
```
Comment (MicPie) [inline data/text_sampling/text_sampling.py:973]: Will we still need that?

File: src/chemnlp/data/convert.py
Comment (MicPie) [inline src/chemnlp/data/convert.py:1]: When do you need those functions? When you set up the data?

File: data/tabular/smiles_to_3d/meta.yaml (hunk 4 above)
Comment (kjappelbaum) [reply data/tabular/smiles_to_3d/meta.yaml]: updated

File: data/text_sampling/text_sampling.py (hunk 7 above)
Comment (kjappelbaum) [reply data/text_sampling/text_sampling.py:973]: this is a debugging comment as all the ones below

File: data/tabular/smiles_to_3d/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol2000__names__noun} of the {#molecule|compound|chemical!} with the {SMILES__description} {SMILES#}?
+      Answer: {mol2000#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {mol3000__names__noun} of the {#molecule|compound|chemical!} with the {SMILES__description} {SMILES#}?
+      Answer: {mol3000#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with {xyz__names__noun} {xyz#}?
+      Answer: {SMILES#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with {mol2000__names__noun} {mol2000#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:93]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with the {mol2000__names__noun} {mol2000#}?
```

File: data/tabular/smiles_to_3d/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+      Answer: {SMILES#}
+    - |-
+      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with {mol3000__names__noun} {mol3000#}?
```
Comment (MicPie) [inline data/tabular/smiles_to_3d/meta.yaml:96]:
```suggestion
      Question: {#What is the|Can you generate the|Can you provide me with the!} {SMILES__description} of the {#molecule|compound|chemical!} with the {mol3000__names__noun} {mol3000#}?
```

**Review by MicPie — APPROVED** [review]
lgtm

**Pull request merged**